### PR TITLE
Issue 405

### PR DESCRIPTION
**Pull request by m8ttyB**
This is for issue #405:
* Moved tests and associated artifacts to the `e2e-tests` directory
* Nudged the README.md file to loosely match the outline of the [fjord project's](https://github.com/mozilla/fjord/blob/master/smoketests/README.rst) readme.

https://github.com/m8ttyB/Socorro-Tests/blob/issue_405/e2e-tests/README.md

@peterbe, @willkg, and @adngdb

/cc @davehunt

### CODE REVIEW

**Comment by willkg** [comment]
Looks fine to me.

File: e2e-tests/README.md [hunk 1]
```diff
@@ -1,120 +1,54 @@
...
-To brush up on Python skills before engaging with us, [Dive Into Python][dive]
-is an excellent resource.  MIT also has [lecture notes on Python][mit] available
-through their open courseware.  The programming concepts you will need to know
-include functions, working with classes, and the basics of object-oriented
-programming.
+* PyPOM
+* pytest
+* pytest-selenium
+* pytest-xdist
+* requests
```
Comment (davehunt) [inline e2e-tests/README.md]: This list of python packages looks a little out of place. Should there be some test associated with them? I wonder if it's actually useful to list them at all though.

File: e2e-tests/README.md [hunk 2]
```diff
@@ -1,120 +1,54 @@
...
-https://quality.mozilla.org/docs/webqa/running-webqa-automated-tests/
-
-This wiki page has some advanced instructions specific to Windows:
 
-https://wiki.mozilla.org/QA_SoftVision_Team/WebQA_Automation
+We suggest using a different virtual environment for these tests than the
+rest of Socorro so you're not mixing requirements:
 
+	$ mkvirtualenv socorro-tests
+	$ pip install -r requirements.txt
```
Comment (davehunt) [inline e2e-tests/README.md]: You might want to include the `cd e2e-tests` to make sure the user is in the correct directory, or include the page in the commands.

File: e2e-tests/README.md [hunk 3]
```diff
@@ -1,120 +1,54 @@
...
-If you have cloned this project already then you can skip this!
-GitHub has excellent guides for [Windows][GitWin], [MacOSX][GitMacOSX], and
-[Linux][GitLinux].
+___Running tests against localhost___
 
-#### Python
-Before you will be able to run these tests you will need to have
-[Python 2.6.8+][Python] installed.
-[Python]: http://www.python.org/download/releases/2.6.8/
+	# run tests against localhost
```
Comment (davehunt) [inline e2e-tests/README.md]: This comment just repeats the heading.

File: e2e-tests/README.md [hunk 4]
```diff
@@ -1,120 +1,54 @@
...
-While most of us have had some experience using virtual machines,
-[virtualenv][venv] is something else entirely.  It's used to keep libraries
-that you install from clashing and messing up your local environment.  After
-installing virtualenv, installing [virtualenvwrapper][wrapper] will give you
-some nice commands to use with virtualenvwrapper. [virtualenv][venv] will allow
-you to install Python modules and run your tests in a sandboxed environment.
+___Running the tests on stage___
 
-__note__
+	# run tests against our staging environment
```
Comment (davehunt) [inline e2e-tests/README.md]: This comment just repeats the heading.

File: e2e-tests/README.md [hunk 5]
```diff
@@ -142,8 +76,75 @@ command line options available. To see the options available, run
...
+version may not be compatible. Upgrading each of them to latest often fixes it.
+ 
+Tips and tricks
+---------------
+
+Because Selenium opens real browser windows, it will steal focus and switch
+workspaces. Firefox doesn't have a headless mode of operation, so we can't 
+simply turn off the UI.
+
+__Use a differnet driver__
```
Comment (davehunt) [inline e2e-tests/README.md]: Typo: different

File: e2e-tests/README.md [hunk 6]
```diff
@@ -142,8 +76,75 @@ command line options available. To see the options available, run
...
+
+[pytest-selenium][pytest-selenium] provides the ability to run tests against other 
+driver executables as well as external providers -- execuatables and providers include PhantomJS, SauceLabs, BrowserStack, etc.
```
Comment (davehunt) [inline e2e-tests/README.md]: Typo: executables

File: e2e-tests/README.md (hunk 1 above)
Comment (m8ttyB) [inline e2e-tests/README.md]: My preference would be to remove this list and add most make mention that there are specific python packages the tests use and suggest the reader review the `requirements.txt` file.

File: e2e-tests/README.md (hunk 6 above)
Comment (davehunt) [inline e2e-tests/README.md]: From Firefox 48 there's also an executable required (GeckoDriver). You could just direct users to the pytest-selenium user guide (which actually needs an update here).

File: e2e-tests/README.md (hunk 2 above)
Comment (m8ttyB) [inline e2e-tests/README.md]: Good catch .. fixed.

**Comment by peterbe** [comment]
Why did you move the files around in this repo? Is that just warm up? I thought we were going to take this whole repo and copy it into github.com/mozilla/socorro/e2e-tests/

**Comment by m8ttyB** [comment]
@peterbe it's a warm up. I thought it also might help pull in the eyes of my team as well as the community.

File: e2e-tests/README.md (hunk 6 above)
Comment (m8ttyB) [inline e2e-tests/README.md]: Is there an eta on when that documentation will land? I'll gladly update the docs here to point to that.

Comment (davehunt) [inline e2e-tests/README.md]: No, though you don't have anything here that isn't already covered and you already link to the documentation. The update is about GeckoDriver for Firefox 48 and above. It's on my list, but fairly low down.

File: e2e-tests/README.md [hunk 7]
```diff
@@ -142,8 +73,74 @@ command line options available. To see the options available, run
...
+workspaces. Firefox doesn't have a headless mode of operation, so we can't 
+simply turn off the UI.
+
+__Use a different driver__
+
+[pytest-selenium] provides the ability to run tests against [many other][test envs] browser environments -- consider using a different driver executable or external provider.
+
+__xvfb__
+
+XCFB provides a fairly easily work around on Linux.
```
Comment (stephendonner) [inline e2e-tests/README.md]: s/XCFB/Xvfb, per https://www.x.org/archive/current/doc/man/man1/Xvfb.1.xhtml

File: e2e-tests/README.md [hunk 8]
```diff
@@ -1,120 +1,51 @@
...
-
-To brush up on Python skills before engaging with us, [Dive Into Python][dive]
-is an excellent resource.  MIT also has [lecture notes on Python][mit] available
-through their open courseware.  The programming concepts you will need to know
-include functions, working with classes, and the basics of object-oriented
-programming.
+To review the specific Python packages the tests utilize, please review `e2e-test/requirements.txt`.
 
-Questions are always welcome
+Setup and run Socorro tests
```
Comment (stephendonner) [inline e2e-tests/README.md]: Uber-nit: s/Setup/Set up/

**Comment by m8ttyB** [comment]
Thank you all for the helpful feedback, v. 0.0.1 of the new readme done. 

Merging.

**Pull request merged**